src/Synapse/Synapse/Models/SynapseAnalyticsRoleClient.cs

```diff
@@ -29,45 +30,51 @@ public SynapseAnalyticsRoleClient(string workspaceName, IAzureContext context)
 
             string suffix = context.Environment.GetEndpoint(AzureEnvironment.ExtendedEndpoint.AzureSynapseAnalyticsEndpointSuffix);
             Uri uri = new Uri("https://" + workspaceName + "." + suffix);
-            _accessControlClient = new AccessControlClient(uri, new AzureSessionCredential(context));
+            _roleAssignmentsClient = new RoleAssignmentsClient(uri, new AzureSessionCredential(context));
+            _roleDefinitionsClient = new RoleDefinitionsClient(uri, new AzureSessionCredential(context));
             _activeDirectoryClient = new ActiveDirectoryClient(context);
         }
 
-        public IReadOnlyList<RoleAssignmentDetails> ListRoleAssignments(string roleDefinitionId = null, string objectId = null, string continuationToken = null)
+        public RoleAssignmentDetailsList ListRoleAssignments(string roleDefinitionId = null, string objectId = null, string continuationToken = null)
         {
-            return _accessControlClient.GetRoleAssignments(roleDefinitionId, objectId, continuationToken).Value;
+            return _roleAssignmentsClient.ListRoleAssignments(roleDefinitionId, objectId, continuationToken).Value;
         }
 
         public RoleAssignmentDetails GetRoleAssignmentById(string roleAssignmentId)
         {
-            return _accessControlClient.GetRoleAssignmentById(roleAssignmentId);
+            return _roleAssignmentsClient.GetRoleAssignmentById(roleAssignmentId);
         }
 
-        public RoleAssignmentDetails CreateRoleAssignment(string roleDefinitionId, string objectId)
+        public RoleAssignmentDetails CreateRoleAssignment(string roleAssignmentId, string RoleDefinitionId, string objectId, string scope)
         {
-            RoleAssignmentOptions roleAssignmentOptions = new RoleAssignmentOptions(roleDefinitionId, objectId);
-            return _accessControlClient.CreateRoleAssignment(roleAssignmentOptions).Value;
+            return _roleAssignmentsClient.CreateRoleAssignment(roleAssignmentId, new Guid(RoleDefinitionId), new Guid(objectId), scope);
         }
 
         public void DeleteRoleAssignmentById(string roleAssignmentId)
         {
-            _accessControlClient.DeleteRoleAssignmentById(roleAssignmentId);
+            _roleAssignmentsClient.DeleteRoleAssignmentById(roleAssignmentId);
         }
 
         public void DeleteRoleAssignmentByName(string roleDefinitionId, string objectId)
         {
-            string roleAssignmentId = roleDefinitionId + "-" + objectId;
-            _accessControlClient.DeleteRoleAssignmentById(roleAssignmentId);
+            var roleAssignment = _roleAssignmentsClient.ListRoleAssignments(roleDefinitionId, objectId, null).Value.Value.Select(element => new PSRoleAssignmentDetails(element));
+            string roleAssignmentId = roleAssignment.SingleOrDefault().RoleAssignmentId;
```

Comment:
>roleAssignment.SingleOrDefault() [](start = 38, length = 32)

Please do consider all possiblities. When such role assignment doesn't exist, code here will throw null reference exception, right? The exception doesn't look friendly to our customers - they cannot determine what is going wrong instantly. You might consider a better error handling way here.

Reply:
![image](https://user-images.githubusercontent.com/46309533/112274498-86014280-8cb9-11eb-817c-df761fc6784b.png)